File: Maximum sum increasing subsequence - GFG/maximum-sum-increasing-subsequence.java
//{ Driver Code Starts
//Initial Template for Java

import java.io.*;
import java.util.*;
class GfG
{
    public static void main(String args[])
        {
            Scanner sc = new Scanner(System.in);
            int t = sc.nextInt();
            while(t-->0)
                {
                    int n = sc.nextInt();
                    int Arr[] = new int[n];
                    for(int i = 0;i<n;i++)
                        Arr[i] = sc.nextInt();
                    Solution ob = new Solution();
                    System.out.println(ob.maxSumIS(Arr,n));
                }
        }
}    
// } Driver Code Ends


//User function Template for Java

class Solution
{
public int maxSumIS(int arr[], int n) {
	    int i, j, max = 0;  
	    int msis[] = new int[n];  
	  
	    /* Initialize msis values  
	    for all indexes */
	    // initializing the msis array with the same values as the input array
	    for ( i = 0; i < n; i++ )  
	        msis[i] = arr[i];  
	  
	    /* Compute maximum sum values  
	    in bottom up manner */
	    // calculating the maximum sum increasing subsequence using dynamic programming
	    for ( i = 1; i < n; i++ )  
	        for ( j = 0; j < i; j++ )  
	            if (arr[i] > arr[j] &&  
	                msis[i] < msis[j] + arr[i])  
	                msis[i] = msis[j] + arr[i];  
	  
	    /* Pick maximum of  
	    all msis values */
	    // finding the maximum value in the msis array
	    for ( i = 0; i < n; i++ )  
	        if ( max < msis[i] )  
	            max = msis[i];  
	  
	    return max;  
	}  
}

File: Maximum sum increasing subsequence - GFG/maximum-sum-increasing-subsequence.js
//{ Driver Code Starts
//Initial Template for javascript

"use strict";

process.stdin.resume();
process.stdin.setEncoding("utf-8");

let inputString = "";
let currentLine = 0;

process.stdin.on("data", (inputStdin) => {
  inputString += inputStdin;
});

process.stdin.on("end", (_) => {
  inputString = inputString
    .trim()
    .split("\n")
    .map((string) => {
      return string.trim();
    });

  main();
});

function readLine() {
  return inputString[currentLine++];
}

/* Function to print an array */
function printArray(arr, size) {
  let i;
  let s = "";
  for (i = 0; i < size; i++) {
    s += arr[i] + " ";
  }
  console.log(s);
}

function main() {
  let t = parseInt(readLine());
  let i = 0;
  for (; i < t; i++) {
    let n = parseInt(readLine());
    let input1 = readLine().split(" ").map((x)=>parseInt(x));
    let arr = new Array(n);
    for(let j = 0;j<n;j++) arr[j] = input1[j];
    let obj = new Solution();
    let res = obj.maxSumIS(arr,n);
    console.log(res);
  }
}
// } Driver Code Ends


//User function Template for javascript

/**
 * @param {number[]} arr
 * @param {number} n
 * @return {number}
 */

class Solution {
    maxSumIS(arr,n){
        let maxSum = 0;
        let msis = new Array(n).fill(0);
      
        for (let i = 0; i < n; i++ ) {
            msis[i] = arr[i];
        }

        for (let i = 1; i < n; i++) {
            for (let j = 0; j < i; j++) {
                if (arr[i] > arr[j] && msis[i] < msis[j] + arr[i]) {
                    msis[i] = msis[j] + arr[i];
                }
            }
        }

        for (let i = 0; i < n; i++) {
            if (maxSum < msis[i]) {
                maxSum = msis[i];
            }
        }

        return maxSum;
    }
}